pkg/ioutil/con_reader.go

```diff
@@ -0,0 +1,159 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package ioutil
+
+import (
+	"errors"
+	"io"
+	"sync"
+)
+
+const defaultBufSize = 32 * 1024
+
+var ErrClosed = errors.New("closed reader")
+
+var bufPool = &sync.Pool{
```

Comment:
would it make sense to initialize `bufPool` when `newConReader` is called, so that you don't need to initialize one with `defaultBufSize` if a custom one is provided?